src/openzaak/components/documenten/tests/test_reserved_document.py

```diff
@@ -0,0 +1,198 @@
+        )
+
+        with self.assertRaises(ReservedDocument.DoesNotExist):
+            ReservedDocument.objects.get(
+                identificatie=self.reserved_identificatie,
+                bronorganisatie=self.bronorganisatie,
+            )
+
+        self.assertEqual(eio.identificatie, self.reserved_identificatie)
+
+    def test_create_enkelvoudig_info_object_with_non_reserved_identificatie(self):
+
+        non_reserved_identificatie = "DOCUMENT-2022-0000900000"
+        url = reverse("reserveddocument-list", kwargs={"version": "1"})
+        data = {
+            "identificatie": non_reserved_identificatie,
+            "bronorganisatie": self.bronorganisatie,
+        }
+
+        response = self.client.post(url, data, format="json")
+        self.assertEqual(response.status_code, 400)
+
+        error_data = response.json()
+        invalid_params = error_data.get("invalidParams", [])
+        self.assertTrue(
+            any(param.get("name") == "identificatie" for param in invalid_params),
+            msg=f"'identificatie' not found in invalidParams: {invalid_params}",
+        )
+
+    def test_create_reservation_when_no_identificatie_provided(self):
+
+        data = {
+            "bronorganisatie": self.bronorganisatie,
+        }
+        response = self.client.post(self.url, data)
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+
+        reserved_doc = ReservedDocument.objects.get(id=response.data["id"])
+
+        self.assertIsNotNone(reserved_doc.identificatie)
+        self.assertTrue(reserved_doc.identificatie.startswith("DOCUMENT-2025"))
+
+    def test_identificatie_generation_excludes_all_taken_ids(self):
+        issued_identificatie = "DOCUMENT-2025-0000000001"
+        EnkelvoudigInformatieObjectFactory.create(
+            identificatie=issued_identificatie,
+            bronorganisatie=self.bronorganisatie,
+        )
+        ReservedDocument.objects.create(
+            identificatie=self.reserved_identificatie,
+            bronorganisatie=self.bronorganisatie,
+        )
+
+        taken_ids = [self.reserved_identificatie, issued_identificatie]
+
+        response = self.client.post(self.url, {"bronorganisatie": self.bronorganisatie})
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+
+        new_id = response.data["identificatie"]
+
+        self.assertNotIn(new_id, taken_ids)
+
+    def test_identificatie_is_generated_for_new_document(self):
+        data = {
+            "bronorganisatie": self.bronorganisatie,
+        }
+
+        response = self.client.post(self.url, data, format="json")
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+
+        identificatie = response.data.get("identificatie")
+        self.assertIsNotNone(identificatie)
+        self.assertTrue(identificatie.startswith("DOCUMENT-" + str(date.today().year)))
+        self.assertEqual(len(identificatie.split("-")[-1]), 10)
```

Comment:
If you use `from freezegun import freeze_time`, you can make this more explicit. You can put a decorator on the entire testcase class like `@freeze_time("2025-01-01T12:00:00")` and then add more readable assertions. I'd prefer to see the exact expected identificatie hardcoded here, e.g. `"DOCUMENT-2025-000000000X"`